# Objective of this notebook is 
1. to identify stocks that have crashed the most eg. MCD Dropped from 217 to 148, whopping 31%
2. Which stocks have risen the most since then (Meaning, might crash the most since economically have not recovered but have been overinflated) 
3. Which are large cap stocks that crashed the most
4. Or by SNP 500
5. Leading to Tracking of PE Ratio

### Other Tracking via
1. PE Ratio
2. Tracking by Market Cap

# Objective of Notebook is to Identify which is the "Best" Stock to Buy When Stock Crashes. 
### Eg. MCD that dropped from 217 to 148, a whopping = 31%

In [1]:
import os
import pandas as pd

In [2]:
datapath = '../obtain_data/ticker_data/ticker_data_Daily_2021-12-11/'
os.listdir(datapath)

['AACG_ended_2021-12-11.csv',
 'AAC_ended_2021-12-11.csv',
 'AAIC_ended_2021-12-11.csv',
 'AAL_ended_2021-12-11.csv',
 'AAMC_ended_2021-12-11.csv',
 'AAME_ended_2021-12-11.csv',
 'AAN_ended_2021-12-11.csv',
 'AAOI_ended_2021-12-11.csv',
 'AAON_ended_2021-12-11.csv',
 'AAPL_ended_2021-12-11.csv',
 'AAP_ended_2021-12-11.csv',
 'AAQC_ended_2021-12-11.csv',
 'AAT_ended_2021-12-11.csv',
 'AAU_ended_2021-12-11.csv',
 'AAWW_ended_2021-12-11.csv',
 'AA_ended_2021-12-11.csv',
 'ABBV_ended_2021-12-11.csv',
 'ABB_ended_2021-12-11.csv',
 'ABCB_ended_2021-12-11.csv',
 'ABCL_ended_2021-12-11.csv',
 'ABCM_ended_2021-12-11.csv',
 'ABC_ended_2021-12-11.csv',
 'ABEO_ended_2021-12-11.csv',
 'ABEV_ended_2021-12-11.csv',
 'ABGI_ended_2021-12-11.csv',
 'ABG_ended_2021-12-11.csv',
 'ABIO_ended_2021-12-11.csv',
 'ABMD_ended_2021-12-11.csv',
 'ABM_ended_2021-12-11.csv',
 'ABNB_ended_2021-12-11.csv',
 'ABR_ended_2021-12-11.csv',
 'ABST_ended_2021-12-11.csv',
 'ABTX_ended_2021-12-11.csv',
 'ABT_ended_2021-12-11.

# Get first DataFrame

In [3]:
df_date = pd.read_csv(datapath + 'AAPL_ended_2021-12-11.csv')
print("df_date:\n",df_date)
df_date = df_date[['Date']]
df_date


df_date:
             Date        Open        High         Low       Close   Adj Close  \
0     1999-12-31    3.605714    3.674142    3.553571    3.672000    3.500993   
1     2000-01-03    3.745714    4.017857    3.631696    3.997767    3.811588   
2     2000-01-04    3.865714    3.951000    3.613839    3.660714    3.490231   
3     2000-01-05    3.705714    3.948660    3.678571    3.714285    3.541308   
4     2000-01-06    3.790000    3.821428    3.392857    3.392857    3.234848   
...          ...         ...         ...         ...         ...         ...   
5518  2021-12-06  164.289993  167.879898  164.279999  165.320007  165.320007   
5519  2021-12-07  169.080002  171.580002  168.339996  171.179993  171.179993   
5520  2021-12-08  172.125000  175.960007  170.699997  175.080002  175.080002   
5521  2021-12-09  174.910004  176.750000  173.919998  174.559998  174.559998   
5522  2021-12-10  175.205002  179.630005  174.690002  179.449997  179.449997   

         Volume  
0      4095

Date
0     1999-12-31
1     2000-01-03
2     2000-01-04
3     2000-01-05
4     2000-01-06
...          ...
5518  2021-12-06
5519  2021-12-07
5520  2021-12-08
5521  2021-12-09
5522  2021-12-10

[5523 rows x 1 columns]

In [4]:
ticker_csv_list = os.listdir(datapath)
ticker_csv_list

['AACG_ended_2021-12-11.csv',
 'AAC_ended_2021-12-11.csv',
 'AAIC_ended_2021-12-11.csv',
 'AAL_ended_2021-12-11.csv',
 'AAMC_ended_2021-12-11.csv',
 'AAME_ended_2021-12-11.csv',
 'AAN_ended_2021-12-11.csv',
 'AAOI_ended_2021-12-11.csv',
 'AAON_ended_2021-12-11.csv',
 'AAPL_ended_2021-12-11.csv',
 'AAP_ended_2021-12-11.csv',
 'AAQC_ended_2021-12-11.csv',
 'AAT_ended_2021-12-11.csv',
 'AAU_ended_2021-12-11.csv',
 'AAWW_ended_2021-12-11.csv',
 'AA_ended_2021-12-11.csv',
 'ABBV_ended_2021-12-11.csv',
 'ABB_ended_2021-12-11.csv',
 'ABCB_ended_2021-12-11.csv',
 'ABCL_ended_2021-12-11.csv',
 'ABCM_ended_2021-12-11.csv',
 'ABC_ended_2021-12-11.csv',
 'ABEO_ended_2021-12-11.csv',
 'ABEV_ended_2021-12-11.csv',
 'ABGI_ended_2021-12-11.csv',
 'ABG_ended_2021-12-11.csv',
 'ABIO_ended_2021-12-11.csv',
 'ABMD_ended_2021-12-11.csv',
 'ABM_ended_2021-12-11.csv',
 'ABNB_ended_2021-12-11.csv',
 'ABR_ended_2021-12-11.csv',
 'ABST_ended_2021-12-11.csv',
 'ABTX_ended_2021-12-11.csv',
 'ABT_ended_2021-12-11.

In [5]:
for i in ticker_csv_list:
    print(i)

AACG_ended_2021-12-11.csv
AAC_ended_2021-12-11.csv
AAIC_ended_2021-12-11.csv
AAL_ended_2021-12-11.csv
AAMC_ended_2021-12-11.csv
AAME_ended_2021-12-11.csv
AAN_ended_2021-12-11.csv
AAOI_ended_2021-12-11.csv
AAON_ended_2021-12-11.csv
AAPL_ended_2021-12-11.csv
AAP_ended_2021-12-11.csv
AAQC_ended_2021-12-11.csv
AAT_ended_2021-12-11.csv
AAU_ended_2021-12-11.csv
AAWW_ended_2021-12-11.csv
AA_ended_2021-12-11.csv
ABBV_ended_2021-12-11.csv
ABB_ended_2021-12-11.csv
ABCB_ended_2021-12-11.csv
ABCL_ended_2021-12-11.csv
ABCM_ended_2021-12-11.csv
ABC_ended_2021-12-11.csv
ABEO_ended_2021-12-11.csv
ABEV_ended_2021-12-11.csv
ABGI_ended_2021-12-11.csv
ABG_ended_2021-12-11.csv
ABIO_ended_2021-12-11.csv
ABMD_ended_2021-12-11.csv
ABM_ended_2021-12-11.csv
ABNB_ended_2021-12-11.csv
ABR_ended_2021-12-11.csv
ABST_ended_2021-12-11.csv
ABTX_ended_2021-12-11.csv
ABT_ended_2021-12-11.csv
ABUS_ended_2021-12-11.csv
AB_ended_2021-12-11.csv
ACAD_ended_2021-12-11.csv
ACAHU_ended_2021-12-11.csv
ACAHW_ended_2021-12-11.csv


PPTA_ended_2021-12-11.csv
PPT_ended_2021-12-11.csv
PRAA_ended_2021-12-11.csv
PRAX_ended_2021-12-11.csv
PRA_ended_2021-12-11.csv
PRCH_ended_2021-12-11.csv
PRDO_ended_2021-12-11.csv
PRFT_ended_2021-12-11.csv
PRFX_ended_2021-12-11.csv
PRGO_ended_2021-12-11.csv
PRGS_ended_2021-12-11.csv
PRG_ended_2021-12-11.csv
PRIM_ended_2021-12-11.csv
PRI_ended_2021-12-11.csv
PRK_ended_2021-12-11.csv
PRLB_ended_2021-12-11.csv
PRLD_ended_2021-12-11.csv
PRMW_ended_2021-12-11.csv
PROF_ended_2021-12-11.csv
PROG_ended_2021-12-11.csv
PROV_ended_2021-12-11.csv
PRO_ended_2021-12-11.csv
PRPB_ended_2021-12-11.csv
PRPC_ended_2021-12-11.csv
PRPH_ended_2021-12-11.csv
PRPL_ended_2021-12-11.csv
PRPO_ended_2021-12-11.csv
PRQR_ended_2021-12-11.csv
PRSRU_ended_2021-12-11.csv
PRSRW_ended_2021-12-11.csv
PRSR_ended_2021-12-11.csv
PRS_ended_2021-12-11.csv
PRTA_ended_2021-12-11.csv
PRTC_ended_2021-12-11.csv
PRTG_ended_2021-12-11.csv
PRTH_ended_2021-12-11.csv
PRTK_ended_2021-12-11.csv
PRTS_ended_2021-12-11.csv
PRTY_ended_2021-1

# Put it all together

In [6]:
for count in range(len(ticker_csv_list)):#[:50]
    
    if count == 0:
        print("count:",count)
        csv_path = os.listdir(datapath)[count] #change
#         print("csv_path:",csv_path)
        ticker = csv_path.split('_ended')[0]
#         print("ticker:",ticker)
        df_current = pd.read_csv(datapath + csv_path)
        df_current_DC = df_current[['Date','Close']]
        df_current_DC.columns = ['Date', ticker]
        df_current_DC
        
        
        join_df = df_date.merge(
            df_current_DC,
            how="left",
            left_on="Date", # col name on df1; df
            right_on="Date", # col name on df2;df_updated
        )
        df = join_df
#         print("df:",df)
    else:
        
        print("count:",count)
        csv_path = os.listdir(datapath)[count] #change
#         print("csv_path:",csv_path)
        ticker = csv_path.split('_ended')[0]
#         print("ticker:",ticker)
        df_current = pd.read_csv(datapath + csv_path)

        df_current_DC = df_current[['Date','Close']]
        df_current_DC.columns = ['Date', ticker]
        df_current_DC
        
        join_df = df_current_DC.merge(
            df,
            how="right",
            left_on="Date", # col name on df1; df
            right_on="Date", # col name on df2;df_updated
        )
#         print("join_df:\n",join_df)
        df = join_df
        

count: 0
count: 1
count: 2
count: 3
count: 4
count: 5
count: 6
count: 7
count: 8
count: 9
count: 10
count: 11
count: 12
count: 13
count: 14
count: 15
count: 16
count: 17
count: 18
count: 19
count: 20
count: 21
count: 22
count: 23
count: 24
count: 25
count: 26
count: 27
count: 28
count: 29
count: 30
count: 31
count: 32
count: 33
count: 34
count: 35
count: 36
count: 37
count: 38
count: 39
count: 40
count: 41
count: 42
count: 43
count: 44
count: 45
count: 46
count: 47
count: 48
count: 49
count: 50
count: 51
count: 52
count: 53
count: 54
count: 55
count: 56
count: 57
count: 58
count: 59
count: 60
count: 61
count: 62
count: 63
count: 64
count: 65
count: 66
count: 67
count: 68
count: 69
count: 70
count: 71
count: 72
count: 73
count: 74
count: 75
count: 76
count: 77
count: 78
count: 79
count: 80
count: 81
count: 82
count: 83
count: 84
count: 85
count: 86
count: 87
count: 88
count: 89
count: 90
count: 91
count: 92
count: 93
count: 94
count: 95
count: 96
count: 97
count: 98
count: 99
count: 100

count: 759
count: 760
count: 761
count: 762
count: 763
count: 764
count: 765
count: 766
count: 767
count: 768
count: 769
count: 770
count: 771
count: 772
count: 773
count: 774
count: 775
count: 776
count: 777
count: 778
count: 779
count: 780
count: 781
count: 782
count: 783
count: 784
count: 785
count: 786
count: 787
count: 788
count: 789
count: 790
count: 791
count: 792
count: 793
count: 794
count: 795
count: 796
count: 797
count: 798
count: 799
count: 800
count: 801
count: 802
count: 803
count: 804
count: 805
count: 806
count: 807
count: 808
count: 809
count: 810
count: 811
count: 812
count: 813
count: 814
count: 815
count: 816
count: 817
count: 818
count: 819
count: 820
count: 821
count: 822
count: 823
count: 824
count: 825
count: 826
count: 827
count: 828
count: 829
count: 830
count: 831
count: 832
count: 833
count: 834
count: 835
count: 836
count: 837
count: 838
count: 839
count: 840
count: 841
count: 842
count: 843
count: 844
count: 845
count: 846
count: 847
count: 848
count: 849

count: 1464
count: 1465
count: 1466
count: 1467
count: 1468
count: 1469
count: 1470
count: 1471
count: 1472
count: 1473
count: 1474
count: 1475
count: 1476
count: 1477
count: 1478
count: 1479
count: 1480
count: 1481
count: 1482
count: 1483
count: 1484
count: 1485
count: 1486
count: 1487
count: 1488
count: 1489
count: 1490
count: 1491
count: 1492
count: 1493
count: 1494
count: 1495
count: 1496
count: 1497
count: 1498
count: 1499
count: 1500
count: 1501
count: 1502
count: 1503
count: 1504
count: 1505
count: 1506
count: 1507
count: 1508
count: 1509
count: 1510
count: 1511
count: 1512
count: 1513
count: 1514
count: 1515
count: 1516
count: 1517
count: 1518
count: 1519
count: 1520
count: 1521
count: 1522
count: 1523
count: 1524
count: 1525
count: 1526
count: 1527
count: 1528
count: 1529
count: 1530
count: 1531
count: 1532
count: 1533
count: 1534
count: 1535
count: 1536
count: 1537
count: 1538
count: 1539
count: 1540
count: 1541
count: 1542
count: 1543
count: 1544
count: 1545
count: 1546
coun

count: 2148
count: 2149
count: 2150
count: 2151
count: 2152
count: 2153
count: 2154
count: 2155
count: 2156
count: 2157
count: 2158
count: 2159
count: 2160
count: 2161
count: 2162
count: 2163
count: 2164
count: 2165
count: 2166
count: 2167
count: 2168
count: 2169
count: 2170
count: 2171
count: 2172
count: 2173
count: 2174
count: 2175
count: 2176
count: 2177
count: 2178
count: 2179
count: 2180
count: 2181
count: 2182
count: 2183
count: 2184
count: 2185
count: 2186
count: 2187
count: 2188
count: 2189
count: 2190
count: 2191
count: 2192
count: 2193
count: 2194
count: 2195
count: 2196
count: 2197
count: 2198
count: 2199
count: 2200
count: 2201
count: 2202
count: 2203
count: 2204
count: 2205
count: 2206
count: 2207
count: 2208
count: 2209
count: 2210
count: 2211
count: 2212
count: 2213
count: 2214
count: 2215
count: 2216
count: 2217
count: 2218
count: 2219
count: 2220
count: 2221
count: 2222
count: 2223
count: 2224
count: 2225
count: 2226
count: 2227
count: 2228
count: 2229
count: 2230
coun

count: 2832
count: 2833
count: 2834
count: 2835
count: 2836
count: 2837
count: 2838
count: 2839
count: 2840
count: 2841
count: 2842
count: 2843
count: 2844
count: 2845
count: 2846
count: 2847
count: 2848
count: 2849
count: 2850
count: 2851
count: 2852
count: 2853
count: 2854
count: 2855
count: 2856
count: 2857
count: 2858
count: 2859
count: 2860
count: 2861
count: 2862
count: 2863
count: 2864
count: 2865
count: 2866
count: 2867
count: 2868
count: 2869
count: 2870
count: 2871
count: 2872
count: 2873
count: 2874
count: 2875
count: 2876
count: 2877
count: 2878
count: 2879
count: 2880
count: 2881
count: 2882
count: 2883
count: 2884
count: 2885
count: 2886
count: 2887
count: 2888
count: 2889
count: 2890
count: 2891
count: 2892
count: 2893
count: 2894
count: 2895
count: 2896
count: 2897
count: 2898
count: 2899
count: 2900
count: 2901
count: 2902
count: 2903
count: 2904
count: 2905
count: 2906
count: 2907
count: 2908
count: 2909
count: 2910
count: 2911
count: 2912
count: 2913
count: 2914
coun

count: 3516
count: 3517
count: 3518
count: 3519
count: 3520
count: 3521
count: 3522
count: 3523
count: 3524
count: 3525
count: 3526
count: 3527
count: 3528
count: 3529
count: 3530
count: 3531
count: 3532
count: 3533
count: 3534
count: 3535
count: 3536
count: 3537
count: 3538
count: 3539
count: 3540
count: 3541
count: 3542
count: 3543
count: 3544
count: 3545
count: 3546
count: 3547
count: 3548
count: 3549
count: 3550
count: 3551
count: 3552
count: 3553
count: 3554
count: 3555
count: 3556
count: 3557
count: 3558
count: 3559
count: 3560
count: 3561
count: 3562
count: 3563
count: 3564
count: 3565
count: 3566
count: 3567
count: 3568
count: 3569
count: 3570
count: 3571
count: 3572
count: 3573
count: 3574
count: 3575
count: 3576
count: 3577
count: 3578
count: 3579
count: 3580
count: 3581
count: 3582
count: 3583
count: 3584
count: 3585
count: 3586
count: 3587
count: 3588
count: 3589
count: 3590
count: 3591
count: 3592
count: 3593
count: 3594
count: 3595
count: 3596
count: 3597
count: 3598
coun

count: 4199
count: 4200
count: 4201
count: 4202
count: 4203
count: 4204
count: 4205
count: 4206
count: 4207
count: 4208
count: 4209
count: 4210
count: 4211
count: 4212
count: 4213
count: 4214
count: 4215
count: 4216
count: 4217
count: 4218
count: 4219
count: 4220
count: 4221
count: 4222
count: 4223
count: 4224
count: 4225
count: 4226
count: 4227
count: 4228
count: 4229
count: 4230
count: 4231
count: 4232
count: 4233
count: 4234
count: 4235
count: 4236
count: 4237
count: 4238
count: 4239
count: 4240
count: 4241
count: 4242
count: 4243
count: 4244
count: 4245
count: 4246
count: 4247
count: 4248
count: 4249
count: 4250
count: 4251
count: 4252
count: 4253
count: 4254
count: 4255
count: 4256
count: 4257
count: 4258
count: 4259
count: 4260
count: 4261
count: 4262
count: 4263
count: 4264
count: 4265
count: 4266
count: 4267
count: 4268
count: 4269
count: 4270
count: 4271
count: 4272
count: 4273
count: 4274
count: 4275
count: 4276
count: 4277
count: 4278
count: 4279
count: 4280
count: 4281
coun

count: 4882
count: 4883
count: 4884
count: 4885
count: 4886
count: 4887
count: 4888
count: 4889
count: 4890
count: 4891
count: 4892
count: 4893
count: 4894
count: 4895
count: 4896
count: 4897
count: 4898
count: 4899
count: 4900
count: 4901
count: 4902
count: 4903
count: 4904
count: 4905
count: 4906
count: 4907
count: 4908
count: 4909
count: 4910
count: 4911
count: 4912
count: 4913
count: 4914
count: 4915
count: 4916
count: 4917
count: 4918
count: 4919
count: 4920
count: 4921
count: 4922
count: 4923
count: 4924
count: 4925
count: 4926
count: 4927
count: 4928
count: 4929
count: 4930
count: 4931
count: 4932
count: 4933
count: 4934
count: 4935
count: 4936
count: 4937
count: 4938
count: 4939
count: 4940
count: 4941
count: 4942
count: 4943
count: 4944
count: 4945
count: 4946
count: 4947
count: 4948
count: 4949
count: 4950
count: 4951
count: 4952
count: 4953
count: 4954
count: 4955
count: 4956
count: 4957
count: 4958
count: 4959
count: 4960
count: 4961
count: 4962
count: 4963
count: 4964
coun

count: 5565
count: 5566
count: 5567
count: 5568
count: 5569
count: 5570
count: 5571
count: 5572
count: 5573
count: 5574
count: 5575
count: 5576
count: 5577
count: 5578
count: 5579
count: 5580
count: 5581
count: 5582
count: 5583
count: 5584
count: 5585
count: 5586
count: 5587
count: 5588
count: 5589
count: 5590
count: 5591
count: 5592
count: 5593
count: 5594
count: 5595
count: 5596
count: 5597
count: 5598
count: 5599
count: 5600
count: 5601
count: 5602
count: 5603
count: 5604
count: 5605
count: 5606
count: 5607
count: 5608
count: 5609
count: 5610
count: 5611
count: 5612
count: 5613
count: 5614
count: 5615
count: 5616
count: 5617
count: 5618
count: 5619
count: 5620
count: 5621
count: 5622
count: 5623
count: 5624
count: 5625
count: 5626
count: 5627
count: 5628
count: 5629
count: 5630
count: 5631
count: 5632
count: 5633
count: 5634
count: 5635
count: 5636
count: 5637
count: 5638
count: 5639
count: 5640
count: 5641
count: 5642
count: 5643
count: 5644
count: 5645
count: 5646
count: 5647
coun

count: 6248
count: 6249
count: 6250
count: 6251
count: 6252
count: 6253
count: 6254
count: 6255
count: 6256
count: 6257
count: 6258
count: 6259
count: 6260
count: 6261
count: 6262
count: 6263
count: 6264
count: 6265
count: 6266
count: 6267
count: 6268
count: 6269
count: 6270
count: 6271
count: 6272
count: 6273
count: 6274
count: 6275
count: 6276
count: 6277
count: 6278
count: 6279
count: 6280
count: 6281
count: 6282
count: 6283
count: 6284
count: 6285
count: 6286
count: 6287
count: 6288
count: 6289
count: 6290
count: 6291
count: 6292
count: 6293
count: 6294
count: 6295
count: 6296
count: 6297
count: 6298
count: 6299
count: 6300
count: 6301
count: 6302
count: 6303
count: 6304
count: 6305
count: 6306
count: 6307
count: 6308
count: 6309
count: 6310
count: 6311
count: 6312
count: 6313
count: 6314
count: 6315
count: 6316
count: 6317
count: 6318
count: 6319
count: 6320
count: 6321
count: 6322
count: 6323
count: 6324
count: 6325
count: 6326
count: 6327
count: 6328
count: 6329
count: 6330
coun

In [7]:
df_short = df.copy()

# Stock Crash of 20 February 2020
### Recovery Date:  27 March 2020

In [8]:
# df_short['Date'] == ['2020-02-20']
df_crashdate = df_short[df_short['Date']== '2020-02-20'].index[0]
df_recoverydate = df_short[df_short['Date']== '2020-03-27'].index[0]
print("df_crashdate:",df_crashdate)
print("df_recoverydate:",df_recoverydate)


df_crashdate: 5065
df_recoverydate: 5091


In [9]:
df_short = df_short[df_crashdate:df_recoverydate]
df_short

Date          Z  ZY   ZYXI  ZYNE       ZYME  ZWRK  ZWRKW  ZWRKU  \
5065  2020-02-20  63.630001 NaN  10.98  4.81  46.439999   NaN    NaN    NaN   
5066  2020-02-21  65.000000 NaN  11.86  4.92  44.630001   NaN    NaN    NaN   
5067  2020-02-24  59.689999 NaN  13.36  4.63  41.480000   NaN    NaN    NaN   
5068  2020-02-25  57.779999 NaN  12.72  4.96  40.419998   NaN    NaN    NaN   
5069  2020-02-26  57.080002 NaN  12.63  4.54  40.450001   NaN    NaN    NaN   
5070  2020-02-27  54.619999 NaN  11.92  4.04  39.700001   NaN    NaN    NaN   
5071  2020-02-28  55.810001 NaN  13.02  4.26  40.930000   NaN    NaN    NaN   
5072  2020-03-02  54.400002 NaN  14.03  4.18  41.340000   NaN    NaN    NaN   
5073  2020-03-03  53.209999 NaN  13.43  4.18  38.549999   NaN    NaN    NaN   
5074  2020-03-04  54.250000 NaN  14.40  4.21  39.040001   NaN    NaN    NaN   
5075  2020-03-05  51.549999 NaN  14.38  4.10  39.049999   NaN    NaN    NaN   
5076  2020-03-06  49.349998 NaN  14.17  4.02  38.119999   NaN    NaN    NaN   
5077  2020-03-09  44.970001 NaN  12.41  3.46  36.410000   NaN    NaN    NaN   
5078  2020-03-10  45.389999 NaN  11.90  3.82  35.200001   NaN    NaN    NaN   
5079  2020-03-11  41.490002 NaN  11.09  3.89  32.259998   NaN    NaN    NaN   
5080  2020-03-12  35.529999 NaN   9.16  3.34  28.740000   NaN    NaN    NaN   
5081  2020-03-13  36.340000 NaN   9.99  3.32  28.969999   NaN    NaN    NaN   
5082  2020-03-16  28.670000 NaN   9.35  2.96  24.280001   NaN    NaN    NaN   
5083  2020-03-17  28.120001 NaN  11.59  3.02  27.580000   NaN    NaN    NaN   
5084  2020-03-18  25.010000 NaN  10.32  2.90  24.760000   NaN    NaN    NaN   
5085  2020-03-19  26.320000 NaN  10.81  3.10  26.309999   NaN    NaN    NaN   
5086  2020-03-20  27.190001 NaN   9.78  3.25  26.879999   NaN    NaN    NaN   
5087  2020-03-23  31.219999 NaN  10.43  3.66  28.260000   NaN    NaN    NaN   
5088  2020-03-24  36.820000 NaN  10.43  3.76  31.770000   NaN    NaN    NaN   
5089  2020-03-25  39.889999 NaN  10.35  3.83  31.680000   NaN    NaN    NaN   
5090  2020-03-26  40.779999 NaN  11.10  3.94  32.759998   NaN    NaN    NaN   

        ZVO  ...       AAPL       AAON    AAOI  AAN    AAME       AAMC  \
5065  2.090  ...  80.074997  56.490002  11.820  NaN  2.2500  21.966900   
5066  1.900  ...  78.262497  55.244999  11.280  NaN  2.6000  21.870001   
5067  1.810  ...  74.544998  53.759998  10.585  NaN  2.5300  25.990000   
5068  1.740  ...  72.019997  53.020000  10.195  NaN  2.4000  28.580000   
5069  1.690  ...  73.162498  53.119999  10.180  NaN  2.5000  25.160000   
5070  1.500  ...  68.379997  57.590000   9.560  NaN  2.1700  23.990000   
5071  1.570  ...  68.339996  55.009998   8.930  NaN  2.4700  21.347000   
5072  1.590  ...  74.702499  57.889999   9.030  NaN  2.6100  20.799999   
5073  1.810  ...  72.330002  56.180000   9.400  NaN  2.5700  20.119499   
5074  1.680  ...  75.684998  59.779999   9.380  NaN  2.5700  21.500000   
5075  1.750  ...  73.230003  58.610001   9.040  NaN  2.5200  23.580000   
5076  1.590  ...  72.257500  58.970001   8.790  NaN  2.5000  23.940001   
5077  1.450  ...  66.542503  51.680000   7.720  NaN  2.5500  20.505899   
5078  1.320  ...  71.334999  51.959999   8.120  NaN  2.6000  19.940800   
5079  1.310  ...  68.857498  44.220001   7.040  NaN  2.5810  19.940001   
5080  1.310  ...  62.057499  41.270000   5.680  NaN  2.5200  17.480000   
5081  1.400  ...  69.492500  52.099998   6.110  NaN  2.5000  15.930000   
5082  1.110  ...  60.552502  51.290001   5.640  NaN  2.4200  14.075800   
5083  2.010  ...  63.215000  47.889999   5.960  NaN  2.2000  15.500000   
5084  1.280  ...  61.667500  47.360001   5.500  NaN  2.2000  10.950000   
5085  1.455  ...  61.195000  49.169998   6.550  NaN  2.4000  11.120000   
5086  1.250  ...  57.310001  45.389999   6.740  NaN  2.3100  11.400000   
5087  1.265  ...  56.092499  47.750000   6.900  NaN  2.2377  10.563500   
5088  1.320  ...  61.720001  45.490002   7.700  NaN  2.4000  10.920000   
5089  1.300  ...  61.380001 

# Section 1. Get Min , Get Max. Calculate largest price drop in %
# Section 2. Get Largest Increase from the min to current date

# Identify Largest PE Ratio Drop


In [10]:
df_ = df_short.copy()
colName = df_.columns.to_list()[1:]

ticker_min_list = []
ticker_max_list = []
print("colName:",colName)
for ticker in colName:
    print("ticker:",ticker)
    ticker_min = min(df_[ticker])
    ticker_max = max(df_[ticker])

    ticker_min_list.append(ticker_min)
    ticker_max_list.append(ticker_max)

colName: ['Z', 'ZY', 'ZYXI', 'ZYNE', 'ZYME', 'ZWRK', 'ZWRKW', 'ZWRKU', 'ZVO', 'ZUO', 'ZUMZ', 'ZTS', 'ZTR', 'ZTO', 'ZS', 'ZSAN', 'ZOM', 'ZNTL', 'ZNTE', 'ZNTEW', 'ZNTEU', 'ZNH', 'ZNGA', 'ZM', 'ZLAB', 'ZKIN', 'ZI', 'ZIXI', 'ZIVO', 'ZIVOW', 'ZIP', 'ZIOP', 'ZION', 'ZIONP', 'ZIONO', 'ZIONL', 'ZIM', 'ZH', 'ZG', 'ZGYH', 'ZGYHW', 'ZGYHR', 'ZGNX', 'ZEV', 'ZEUS', 'ZEPP', 'ZEN', 'ZEAL', 'ZDGE', 'ZCMD', 'ZBRA', 'ZBH', 'Y', 'YY', 'YVR', 'YUM', 'YUMC', 'YTRA', 'YTPG', 'YTEN', 'YSG', 'YSAC', 'YSACW', 'YSACU', 'YRD', 'YQ', 'YPF', 'YORW', 'YNDX', 'YMTX', 'YMAB', 'YJ', 'YI', 'YGMZ', 'YEXT', 'YETI', 'YELP', 'YELL', 'YCBD', 'YALA', 'YAC', 'X', 'XYL', 'XYF', 'XXII', 'XTNT', 'XTLB', 'XSPA', 'XRX', 'XRAY', 'XP', 'XPO', 'XPOA', 'XPL', 'XPEV', 'XPER', 'XPEL', 'XPDI', 'XPDIW', 'XPDIU', 'XONE', 'XOM', 'XOMA', 'XOMAP', 'XOMAO', 'XOG', 'XNET', 'XNCR', 'XM', 'XL', 'XLRN', 'XLNX', 'XIN', 'XHR', 'XGN', 'XFOR', 'XFLT', 'XERS', 'XENT', 'XENE', 'XEL', 'XELB', 'XELA', 'XCUR', 'XBIT', 'XBIO', 'XBIOW', 'XAIR', 'W', 'WY', 'W

ticker: THR
ticker: THRY
ticker: THRM
ticker: THQ
ticker: THO
ticker: THM
ticker: THMO
ticker: THMA
ticker: THMAW
ticker: THMAU
ticker: THG
ticker: THFF
ticker: THC
ticker: THCA
ticker: THCAW
ticker: TG
ticker: TGT
ticker: TGTX
ticker: TGS
ticker: TGP
ticker: TGNA
ticker: TGLS
ticker: TGI
ticker: TGH
ticker: TGB
ticker: TGA
ticker: TFX
ticker: TFSL
ticker: TFSA
ticker: TFII
ticker: TFFP
ticker: TFC
ticker: TEX
ticker: TEVA
ticker: TETC
ticker: TETCW
ticker: TETCU
ticker: TESS
ticker: TER
ticker: TERN
ticker: TEO
ticker: TEN
ticker: TENX
ticker: TENB
ticker: TEL
ticker: TELL
ticker: TELA
ticker: TEKK
ticker: TEKKW
ticker: TEKKU
ticker: TEI
ticker: TEF
ticker: TEDU
ticker: TECTP
ticker: TECK
ticker: TECH
ticker: TEAM
ticker: TEAF
ticker: TD
ticker: TDY
ticker: TDW
ticker: TDUP
ticker: TDS
ticker: TDOC
ticker: TDG
ticker: TDF
ticker: TDC
ticker: TDA
ticker: TDAC
ticker: TDACW
ticker: TDACU
ticker: TC
ticker: TCX
ticker: TCVA
ticker: TCS
ticker: TCRR
ticker: TCPC
ticker: TCON
ticker: TCOM


ticker: PBI
ticker: PBIP
ticker: PBH
ticker: PBHC
ticker: PBF
ticker: PBFX
ticker: PBFS
ticker: PBC
ticker: PBCT
ticker: PBCTP
ticker: PBA
ticker: PAY
ticker: PAYX
ticker: PAYS
ticker: PAYC
ticker: PAYA
ticker: PAX
ticker: PAVM
ticker: PAVMZ
ticker: PAVMW
ticker: PATK
ticker: PATI
ticker: PATH
ticker: PASG
ticker: PAR
ticker: PARR
ticker: PAQC
ticker: PAQCW
ticker: PAQCU
ticker: PANW
ticker: PANL
ticker: PANA
ticker: PAM
ticker: PALI
ticker: PAI
ticker: PAIC
ticker: PAICW
ticker: PAHC
ticker: PAG
ticker: PAGS
ticker: PAGP
ticker: PAE
ticker: PAEWW
ticker: PAC
ticker: PACX
ticker: PACXW
ticker: PACXU
ticker: PACW
ticker: PACK
ticker: PACB
ticker: PAA
ticker: PAAS
ticker: O
ticker: OZON
ticker: OZK
ticker: OYST
ticker: OXY
ticker: OXSQ
ticker: OXSQZ
ticker: OXSQL
ticker: OXSQG
ticker: OXM
ticker: OXLC
ticker: OXLCP
ticker: OXLCM
ticker: OXLCL
ticker: OXBR
ticker: OXBRW
ticker: OWL
ticker: OVV
ticker: OVLY
ticker: OVID
ticker: OVBC
ticker: OUT
ticker: OUST
ticker: OTTR
ticker: OTRK
ticker

ticker: LSPD
ticker: LSI
ticker: LSF
ticker: LSEA
ticker: LSEAW
ticker: LSCC
ticker: LSBK
ticker: LRN
ticker: LRMR
ticker: LRCX
ticker: LQDT
ticker: LQDA
ticker: LPX
ticker: LPTX
ticker: LPTH
ticker: LPSN
ticker: LPRO
ticker: LPL
ticker: LPLA
ticker: LPI
ticker: LPG
ticker: LPCN
ticker: LOW
ticker: LOV
ticker: LOVE
ticker: LOTZ
ticker: LOTZW
ticker: LORL
ticker: LOPE
ticker: LOOP
ticker: LOMA
ticker: LOKM
ticker: LOKB
ticker: LOGI
ticker: LOGC
ticker: LODE
ticker: LOCO
ticker: LOB
ticker: LOAN
ticker: LNT
ticker: LNTH
ticker: LNSR
ticker: LNN
ticker: LNG
ticker: LNFA
ticker: LND
ticker: LNDC
ticker: LNC
ticker: LMT
ticker: LMST
ticker: LMRK
ticker: LMRKP
ticker: LMRKO
ticker: LMRKN
ticker: LMPX
ticker: LMNR
ticker: LMNL
ticker: LMND
ticker: LMFA
ticker: LMB
ticker: LMAT
ticker: LMAO
ticker: LMAOW
ticker: LMAOU
ticker: LMACW
ticker: LMACU
ticker: LMACA
ticker: LL
ticker: LLY
ticker: LLNW
ticker: LKQ
ticker: LKFN
ticker: LKCO
ticker: LJPC
ticker: LJAQW
ticker: LJAQU
ticker: LI
ticker: LI

ticker: HGV
ticker: HGSH
ticker: HGLB
ticker: HGH
ticker: HGEN
ticker: HGBL
ticker: HFWA
ticker: HFRO
ticker: HFFG
ticker: HFC
ticker: HFBL
ticker: HE
ticker: HEXO
ticker: HES
ticker: HESM
ticker: HERA
ticker: HERAW
ticker: HERAU
ticker: HEQ
ticker: HEP
ticker: HEPA
ticker: HELE
ticker: HEI
ticker: HEES
ticker: HEAR
ticker: HD
ticker: HDSN
ticker: HDB
ticker: HCXZ
ticker: HCXY
ticker: HCSG
ticker: HCM
ticker: HCKT
ticker: HCI
ticker: HCII
ticker: HCIIU
ticker: HCIC
ticker: HCICW
ticker: HCICU
ticker: HCDI
ticker: HCC
ticker: HCCI
ticker: HCCC
ticker: HCCCW
ticker: HCCCU
ticker: HCA
ticker: HCAT
ticker: HCAR
ticker: HCARW
ticker: HCARU
ticker: HCAQ
ticker: HBT
ticker: HBP
ticker: HBNC
ticker: HBM
ticker: HBMD
ticker: HBI
ticker: HBIO
ticker: HBCP
ticker: HBB
ticker: HBAN
ticker: HBANP
ticker: HBANN
ticker: HA
ticker: HAYW
ticker: HAYN
ticker: HAS
ticker: HASI
ticker: HARP
ticker: HAPP
ticker: HAL
ticker: HALO
ticker: HALL
ticker: HAIN
ticker: HAFC
ticker: HAE
ticker: HAAC
ticker: HAACW


ticker: EMKR
ticker: EMF
ticker: EME
ticker: EMD
ticker: EMCF
ticker: EMAN
ticker: EL
ticker: ELY
ticker: ELYS
ticker: ELVT
ticker: ELTK
ticker: ELS
ticker: ELSE
ticker: ELP
ticker: ELOX
ticker: ELMD
ticker: ELLO
ticker: ELF
ticker: ELDN
ticker: ELC
ticker: ELA
ticker: ELAT
ticker: ELAN
ticker: EKSO
ticker: EJH
ticker: EJFA
ticker: EJFAW
ticker: EJFAU
ticker: EIX
ticker: EIM
ticker: EIG
ticker: EIGR
ticker: EIC
ticker: EH
ticker: EHTH
ticker: EHI
ticker: EHC
ticker: EGY
ticker: EGRX
ticker: EGP
ticker: EGO
ticker: EGLX
ticker: EGLE
ticker: EGHT
ticker: EGF
ticker: EGBN
ticker: EGAN
ticker: EFX
ticker: EFT
ticker: EFSC
ticker: EFR
ticker: EFOI
ticker: EFL
ticker: EFC
ticker: EEX
ticker: EEIQ
ticker: EEFT
ticker: EEA
ticker: ED
ticker: EDU
ticker: EDUC
ticker: EDTX
ticker: EDTXW
ticker: EDTXU
ticker: EDTK
ticker: EDSA
ticker: EDR
ticker: EDRY
ticker: EDN
ticker: EDI
ticker: EDIT
ticker: EDF
ticker: EDD
ticker: EDAP
ticker: EC
ticker: ECPG
ticker: ECOR
ticker: ECOM          
ticker: ECOL


ticker: BYTS
ticker: BYTSW
ticker: BYTSU
ticker: BYSI
ticker: BYRN
ticker: BYND
ticker: BYM
ticker: BYFC
ticker: BYD
ticker: BX
ticker: BXS
ticker: BXRX
ticker: BXP
ticker: BXMX
ticker: BXMT
ticker: BXC
ticker: BW
ticker: BWXT
ticker: BWSN
ticker: BWMX
ticker: BWMN
ticker: BWG
ticker: BWFG
ticker: BWEN
ticker: BWB
ticker: BWA
ticker: BWAY
ticker: BWAC
ticker: BWACW
ticker: BWACU
ticker: BV
ticker: BVXV
ticker: BVS
ticker: BVN
ticker: BVH
ticker: BUSE
ticker: BUR
ticker: BURL
ticker: BUI
ticker: BUD
ticker: BTZ
ticker: BTX
ticker: BTWN
ticker: BTWNW
ticker: BTWNU
ticker: BTU
ticker: BTT
ticker: BTRS
ticker: BTRSW
ticker: BTO
ticker: BTN
ticker: BTNB
ticker: BTI
ticker: BTG
ticker: BTCM
ticker: BTBT
ticker: BTA
ticker: BTAQ
ticker: BTAQW
ticker: BTAQU
ticker: BTAI
ticker: BSY
ticker: BSX
ticker: BSVN
ticker: BST
ticker: BSTZ
ticker: BSRR
ticker: BSQR
ticker: BSN
ticker: BSM
ticker: BSMX
ticker: BSL
ticker: BSKYU
ticker: BSIG
ticker: BSGM
ticker: BSET
ticker: BSBR
ticker: BSBK
ticker: BSA

ticker: ACAHU
ticker: ACAD
ticker: AB
ticker: ABUS
ticker: ABT
ticker: ABTX
ticker: ABST
ticker: ABR
ticker: ABNB
ticker: ABM
ticker: ABMD
ticker: ABIO
ticker: ABG
ticker: ABGI
ticker: ABEV
ticker: ABEO
ticker: ABC
ticker: ABCM
ticker: ABCL
ticker: ABCB
ticker: ABB
ticker: ABBV
ticker: AA
ticker: AAWW
ticker: AAU
ticker: AAT
ticker: AAQC
ticker: AAP
ticker: AAPL
ticker: AAON
ticker: AAOI
ticker: AAN
ticker: AAME
ticker: AAMC
ticker: AAL
ticker: AAIC
ticker: AAC
ticker: AACG


In [11]:
print("ticker_min_list:",ticker_min_list)
print("ticker_max_list:",ticker_max_list)

ticker_min_list: [25.01000022888184, nan, 9.15999984741211, 2.900000095367432, 24.280000686645508, nan, nan, nan, 1.1100000143051147, 7.190000057220459, 17.010000228881836, 92.66000366210938, 5.050000190734863, 23.32999992370605, 39.5, 0.4300000071525574, 0.1492000073194503, nan, nan, nan, nan, 18.950000762939453, 5.869999885559082, 101.76000213623048, 44.540000915527344, 0.6899999976158142, nan, 3.0299999713897705, 0.0799999982118606, nan, nan, 1.899999976158142, 24.07999992370605, 16.5, 19.43000030517578, 22.020000457763672, nan, nan, 23.51000022888184, nan, nan, nan, 17.799999237060547, nan, 9.1899995803833, 12.470000267028809, 54.7400016784668, 23.86000061035156, 0.7900000214576721, nan, 161.50999450683594, 80.62999725341797, 452.8399963378906, 42.0099983215332, 1.8799999952316284, 56.52000045776367, 39.2400016784668, 0.8600000143051147, nan, 3.7699999809265137, nan, nan, nan, nan, 2.990000009536743, nan, 2.569999933242798, 35.599998474121094, 28.90999984741211, nan, 14.39000034332

# Calculation

In [12]:
Low_from_High_in_percentage = [min_ / max_ for min_, max_ in zip(ticker_min_list, ticker_max_list) ] #get Low (%) from ATH 
print("Low_from_High_in_percentage:",Low_from_High_in_percentage)
value_lost_in_percentage = [1 - low_ for low_ in Low_from_High_in_percentage] # Value Lost in % i.e 1- (get Low (%) from ATH)
print("value_lost_in_percentage:",value_lost_in_percentage)

Low_from_High_in_percentage: [0.3847692342904898, nan, 0.6361111173659197, 0.584677434085435, 0.5228251809783944, nan, nan, nan, 0.5311005071243239, 0.4471393091625921, 0.5267884715575282, 0.6451748141789238, 0.4225941649655484, 0.8422382411960024, 0.6060141119217424, 0.43438732217204895, 0.46625003329478254, nan, nan, nan, nan, 0.6356927575456562, 0.8118948644638198, 0.6377538461597312, 0.7091227502260061, 0.5557792883488789, nan, 0.33499170213730595, 0.47169811652304205, nan, nan, 0.5757575768522433, 0.5206486469990498, 0.6615877965787489, 0.6815152630371559, 0.7012739084483578, nan, nan, 0.35843880990278953, nan, nan, nan, 0.5756791459618278, nan, 0.6777285616517019, 0.8564560376130188, 0.6199320474131861, 0.5836794798097811, 0.5064102887401931, nan, 0.6654169153533266, 0.5090599000197201, 0.5657459121741282, 0.6859895031045493, 0.5784615369943472, 0.5441417056246638, 0.8521173144100047, 0.23888889919092654, nan, 0.5046854199215166, nan, nan, nan, nan, 0.6064908952632465, nan, 0.264

In [13]:
Low_from_High_in_percentage_df = pd.DataFrame(Low_from_High_in_percentage, columns = ['Low_from_High_in_percentage'])
value_lost_in_percentage_df = pd.DataFrame(value_lost_in_percentage, columns = ['value_lost_in_percentage'])
ticker_min_list_df = pd.DataFrame(ticker_min_list, columns = ['ticker_min_list'])
ticker_max_list_df = pd.DataFrame(ticker_max_list, columns = ['ticker_max_list'])
df_colName = pd.DataFrame(colName, columns = ['colName'])

df_concat = pd.concat([df_colName,ticker_min_list_df,ticker_max_list_df,Low_from_High_in_percentage_df,value_lost_in_percentage_df], axis=1)
df_concat

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1         ZY              NaN              NaN                          NaN   
2       ZYXI         9.160000        14.400000                     0.636111   
3       ZYNE         2.900000         4.960000                     0.584677   
4       ZYME        24.280001        46.439999                     0.522825   
...      ...              ...              ...                          ...   
6777    AAMC        10.563500        28.580000                     0.369612   
6778     AAL        10.250000        28.510000                     0.359523   
6779    AAIC         1.700000         6.690000                     0.254111   
6780     AAC              NaN              NaN                          NaN   
6781    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  
0                     0.615231  
1                          NaN  
2                     0.363889  
3                     0.415323  
4                     0.477175  
...                        ...  
6777                  0.630388  
6778                  0.640477  
6779                  0.745889  
6780                       NaN  
6781                  0.411684  

[6782 rows x 5 columns]

In [14]:
df_checkpoint1 = df_concat.copy()
dfcp = df_checkpoint1
dfcp

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1         ZY              NaN              NaN                          NaN   
2       ZYXI         9.160000        14.400000                     0.636111   
3       ZYNE         2.900000         4.960000                     0.584677   
4       ZYME        24.280001        46.439999                     0.522825   
...      ...              ...              ...                          ...   
6777    AAMC        10.563500        28.580000                     0.369612   
6778     AAL        10.250000        28.510000                     0.359523   
6779    AAIC         1.700000         6.690000                     0.254111   
6780     AAC              NaN              NaN                          NaN   
6781    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  
0                     0.615231  
1                          NaN  
2                     0.363889  
3                     0.415323  
4                     0.477175  
...                        ...  
6777                  0.630388  
6778                  0.640477  
6779                  0.745889  
6780                       NaN  
6781                  0.411684  

[6782 rows x 5 columns]

In [15]:
dfcp.dropna(axis=0, inplace = True)
dfcp

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
2       ZYXI         9.160000        14.400000                     0.636111   
3       ZYNE         2.900000         4.960000                     0.584677   
4       ZYME        24.280001        46.439999                     0.522825   
8        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
6776    AAME         2.120200         2.610000                     0.812337   
6777    AAMC        10.563500        28.580000                     0.369612   
6778     AAL        10.250000        28.510000                     0.359523   
6779    AAIC         1.700000         6.690000                     0.254111   
6781    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  
0                     0.615231  
2                     0.363889  
3                     0.415323  
4                     0.477175  
8                     0.468899  
...                        ...  
6776                  0.187663  
6777                  0.630388  
6778                  0.640477  
6779                  0.745889  
6781                  0.411684  

[5095 rows x 5 columns]

In [16]:
dfcp.sort_values(by = 'value_lost_in_percentage',ascending = False)

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
5200    CUBA         0.000000         6.524400                     0.000000   
2875     MFA         0.360000         8.030000                     0.044832   
2492    NINE         0.380000         5.810000                     0.065404   
1736    PVAC         1.130000        17.180000                     0.065774   
1503    ROCC         1.130000        17.180000                     0.065774   
...      ...              ...              ...                          ...   
3617    IMCC         0.100000         0.100000                     1.000000   
2182     OPT        21.360001        21.360001                     1.000000   
2161    OSAT         2.820000         2.820000                     1.000000   
3542    IPDN         0.910000         0.910000                     1.000000   
6564    AGTI         9.000000         9.000000                     1.000000   

      value_lost_in_percentage  
5200                  1.000000  
2875                  0.955168  
2492                  0.934596  
1736                  0.934226  
1503                  0.934226  
...                        ...  
3617                  0.000000  
2182                  0.000000  
2161                  0.000000  
3542                  0.000000  
6564                  0.000000  

[5095 rows x 5 columns]

# Section 1 completed:
#### We have sucessfully Identified which stocks collapsed the most

In [17]:
dfcp_sorted_by_valuelost = dfcp.sort_values(by = 'value_lost_in_percentage',ascending = False)
# dfcp.sort_values(by = 'value_lost_in_percentage',ascending = False)
dfcp_sorted_by_valuelost

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
5200    CUBA         0.000000         6.524400                     0.000000   
2875     MFA         0.360000         8.030000                     0.044832   
2492    NINE         0.380000         5.810000                     0.065404   
1736    PVAC         1.130000        17.180000                     0.065774   
1503    ROCC         1.130000        17.180000                     0.065774   
...      ...              ...              ...                          ...   
3617    IMCC         0.100000         0.100000                     1.000000   
2182     OPT        21.360001        21.360001                     1.000000   
2161    OSAT         2.820000         2.820000                     1.000000   
3542    IPDN         0.910000         0.910000                     1.000000   
6564    AGTI         9.000000         9.000000                     1.000000   

      value_lost_in_percentage  
5200                  1.000000  
2875                  0.955168  
2492                  0.934596  
1736                  0.934226  
1503                  0.934226  
...                        ...  
3617                  0.000000  
2182                  0.000000  
2161                  0.000000  
3542                  0.000000  
6564                  0.000000  

[5095 rows x 5 columns]

#### We can see CUBA MFA NINE lost the most, but I for one dont know much about these stocks. So i want to target larger cap stocks eg. 

# Section 2: 
2. Which stocks have risen the most since then (Meaning, might crash the most since economically have not recovered but have been overinflated) --> Leading to Tracking of PE Ratio
- Or Which Stocks has:
  - i. Recovered the most from ATL
  - ii. Has risen the most from Feb 2020 onwards, meaning Ceteris Paribus; disregarding the Crash i.e COVID never happened. 

In [18]:
dfcp.head()
dfcp['colName'].to_list()

['Z',
 'ZYXI',
 'ZYNE',
 'ZYME',
 'ZVO',
 'ZUO',
 'ZUMZ',
 'ZTS',
 'ZTR',
 'ZTO',
 'ZS',
 'ZSAN',
 'ZOM',
 'ZNH',
 'ZNGA',
 'ZM',
 'ZLAB',
 'ZKIN',
 'ZIXI',
 'ZIVO',
 'ZIOP',
 'ZION',
 'ZIONP',
 'ZIONO',
 'ZIONL',
 'ZG',
 'ZGNX',
 'ZEUS',
 'ZEPP',
 'ZEN',
 'ZEAL',
 'ZDGE',
 'ZBRA',
 'ZBH',
 'Y',
 'YY',
 'YVR',
 'YUM',
 'YUMC',
 'YTRA',
 'YTEN',
 'YRD',
 'YPF',
 'YORW',
 'YNDX',
 'YMAB',
 'YJ',
 'YI',
 'YEXT',
 'YETI',
 'YELP',
 'YELL',
 'YCBD',
 'X',
 'XYL',
 'XYF',
 'XXII',
 'XTNT',
 'XTLB',
 'XSPA',
 'XRX',
 'XRAY',
 'XP',
 'XPO',
 'XPL',
 'XPER',
 'XPEL',
 'XONE',
 'XOM',
 'XOMA',
 'XNET',
 'XNCR',
 'XLRN',
 'XLNX',
 'XIN',
 'XHR',
 'XGN',
 'XFOR',
 'XFLT',
 'XERS',
 'XENT',
 'XENE',
 'XEL',
 'XELB',
 'XELA',
 'XCUR',
 'XBIT',
 'XBIO',
 'XBIOW',
 'XAIR',
 'W',
 'WY',
 'WYY',
 'WYNN',
 'WW',
 'WWW',
 'WWR',
 'WWE',
 'WWD',
 'WVVI',
 'WVVIP',
 'WVFC',
 'WVE',
 'WU',
 'WTT',
 'WTTR',
 'WTS',
 'WTRU',
 'WTRH',
 'WTRG',
 'WTREP',
 'WTM',
 'WTI',
 'WTFC',
 'WTFCM',
 'WTER',
 'WTBA',
 'WST

In [19]:
# Get Current Price
curren_price_list = []
for dfcp_ticker in dfcp['colName'].to_list():
    print(dfcp_ticker)
#     dfcp_ticker = 'AAPL'
    temp_df = pd.read_csv('../obtain_data/ticker_data/ticker_data_Daily_2021-12-11/{}_ended_2021-12-11.csv'.format(dfcp_ticker))
    temp_df
    closing_price = temp_df.loc[len(temp_df)-1,'Close'] # Get Current Closing Price
    curren_price_list.append(closing_price)

Z
ZYXI
ZYNE
ZYME
ZVO
ZUO
ZUMZ
ZTS
ZTR
ZTO
ZS
ZSAN
ZOM
ZNH
ZNGA
ZM
ZLAB
ZKIN
ZIXI
ZIVO
ZIOP
ZION
ZIONP
ZIONO
ZIONL
ZG
ZGNX
ZEUS
ZEPP
ZEN
ZEAL
ZDGE
ZBRA
ZBH
Y
YY
YVR
YUM
YUMC
YTRA
YTEN
YRD
YPF
YORW
YNDX
YMAB
YJ
YI
YEXT
YETI
YELP
YELL
YCBD
X
XYL
XYF
XXII
XTNT
XTLB
XSPA
XRX
XRAY
XP
XPO
XPL
XPER
XPEL
XONE
XOM
XOMA
XNET
XNCR
XLRN
XLNX
XIN
XHR
XGN
XFOR
XFLT
XERS
XENT
XENE
XEL
XELB
XELA
XCUR
XBIT
XBIO
XBIOW
XAIR
W
WY
WYY
WYNN
WW
WWW
WWR
WWE
WWD
WVVI
WVVIP
WVFC
WVE
WU
WTT
WTTR
WTS
WTRU
WTRH
WTRG
WTREP
WTM
WTI
WTFC
WTFCM
WTER
WTBA
WST
WSTG
WSR
WSO
WSM
WSFS
WSC
WSBF
WSBC
WRN
WRLD
WRK
WRE
WRB
WRAP
WPRT
WPP
WPM
WPC
WOW
WOR
WORX
WNS
WNEB
WNC
WM
WMT
WMS
WMPN
WMK
WMC
WMB
WLTW
WLMS
WLL
WLK
WLKP
WLFC
WLDN
WK
WKHS
WKEY
WIX
WIW
WIT
WISA
WIRE
WINT
WING
WINA
WILC
WIA
WH
WHR
WHLR
WHLRP
WHLRD
WHLM
WHG
WHF
WHFBZ
WHD
WGO
WF
WFG
WFC
WFCF
WEYS
WEX
WETF
WES
WERN
WEN
WELL
WEI
WEC
WEA
WD
WDFC
WDC
WDAY
WCN
WCC
WB
WBT
WBS
WBK
WBA
WAT
WATT
WASH
WAL
WALA
WAFU
WAFD
WAB
WABC
V
VYNE
VYGR
VXRT
VWTR
VVV
VVR
VVPR
VVNT
VVI
VUZ

NTR
NTRS
NTRSO
NTRA
NTP
NTNX
NTLA
NTIP
NTIC
NTG
NTGR
NTES
NTCT
NTCO
NTB
NTAP
NS
NSYS
NSTG
NSS
NSSC
NSP
NSPR
NSL
NSIT
NSEC
NSC
NSA
NR
NRZ
NRUC
NRT
NRP
NRO
NRK
NRIM
NRG
NRGX
NREF
NRC
NRBO
NQP
NP
NPV
NPTN
NPO
NPK
NOW
NOV
NOVT
NOVN
NOVA
NOTV
NOM
NOMD
NOK
NOG
NODK
NOC
NOA
NOAH
NNY
NNVC
NNN
NNI
NNDM
NNBR
NNA
NM
NMZ
NMT
NMTR
NMTC
NMS
NMR
NMRK
NMRD
NMM
NML
NMI
NMIH
NMG
NMFC
NMCO
NL
NLY
NLTX
NLS
NLSN
NLOK
NKX
NKTR
NKSH
NKG
NKE
NJR
NI
NIU
NISN
NIQ
NIO
NINE
NIM
NIE
NID
NICK
NICE
NH
NHTC
NHS
NHI
NHF
NHC
NG
NGVT
NGVC
NGS
NGM
NGL
NGG
NGD
NFLX
NFJ
NFH
NFG
NFE
NFBK
NEX
NEXT
NEXA
NEW
NEWT
NEWTL
NEWR
NEWP
NEV
NEU
NET
NETI
NETE
NES
NESR
NERV
NEP
NEPT
NEPH
NEO
NEON
NEOG
NEN
NEM
NEE
NEA
NDSN
NDRA
NDP
NDLS
NDAQ
NC
NCZ
NCV
NCTY
NCSM
NCR
NCNA
NCMI
NCLH
NCBS
NCA
NBY
NBW
NBTB
NBSE
NBR
NBRV
NBO
NBN
NBIX
NBH
NBHC
NBEV
NBB
NAZ
NAVI
NAVB
NAT
NATR
NATI
NATH
NAOV
NAN
NAK
NAKD
NAII
NAD
NAC
M
MYSZ
MYRG
MYO
MYOV
MYN
MYJ
MYI
MYGN
MYFW
MYE
MYD
MYC
MX
MXL
MXF
MXE
MXC
MWA
MVT
MVO
MVIS
MVF
MVBF
MU
MUX
MUSA
MUR
MUJ
MUI
MUFG
M

EHTH
EHI
EHC
EGY
EGRX
EGP
EGO
EGLX
EGLE
EGHT
EGF
EGBN
EGAN
EFX
EFT
EFSC
EFR
EFOI
EFL
EFC
EEX
EEFT
EEA
ED
EDU
EDUC
EDSA
EDRY
EDN
EDI
EDIT
EDF
EDD
EDAP
EC
ECPG
ECOR
ECOM          
ECOL
ECL
ECHO
ECF
ECCY
ECCX
ECCB
ECC           
EB
EBTC
EBS
EBSB
EBR
EBMT
EBIX
EBF
EBAY
EA
EAT
EAST
EARN
EAF
EAD
D
DZSI
DY
DYNT
DYAI
DX
DXYN
DXR
DXPE
DXF
DXC
DXCM
DWSN
DVN
DVD
DVA
DVAX
DUO
DUOT
DUK
DUKH
DUKB
DT
DTY
DTW
DTST
DTSS
DTP
DTJ
DTIL
DTF
DTE
DTEA
DS
DSX
DSWL
DSU
DSS
DSPG
DSM
DSL
DSKE
DSGX
DRUA
DRTT
DRRX
DRQ
DRNA
DRI
DRIO
DRH
DRE
DRD
DQ
DPZ
DPW
DPG
DOYU
DOX
DOW
DOV
DORM
DOOR
DOOO
DOMO
DOGZ
DOC
DOCU
DNP
DNOW
DNN
DNLI
DMTK
DMS
DMRC
DMO
DMLP
DMF
DMB
DMAC
DLX
DLTR
DLTH
DLR
DLPN
DLNG
DLHC
DLB
DLA
DK
DKS
DKL
DJCO
DIT
DIS
DISH
DISCK
DISCB
DISCA
DIOD
DIN
DIAX
DHY
DHX
DHT
DHR
DHI
DHIL
DHF
DHC
DHCNI
DG
DGX
DGLY
DGII
DGICB
DGICA
DFS
DFP
DFIN
DFFN
DE
DEX
DESP
DEO
DENN
DELL
DEI
DECK
DEA
DD
DDT
DDS
DDOG
DDF
DDD
DCUE
DCTH
DCP
DCPH
DCO
DCOM
DCOMP
DCI
DCF
DB
DBX
DBVT
DBL
DBI
DBD
DAVA
DAR
DARE
DAO
DAN
DAL
DAKT
DAIO
DAC
C


In [20]:
df_curren_price_list= pd.DataFrame(curren_price_list, columns = ['current_price'])
df_curren_price_list
dfcp.reset_index(drop = True, inplace=True)
dfcp
dfcp1 = pd.concat([dfcp,df_curren_price_list ],axis=1)
dfcp1

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1       ZYXI         9.160000        14.400000                     0.636111   
2       ZYNE         2.900000         4.960000                     0.584677   
3       ZYME        24.280001        46.439999                     0.522825   
4        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
5090    AAME         2.120200         2.610000                     0.812337   
5091    AAMC        10.563500        28.580000                     0.369612   
5092     AAL        10.250000        28.510000                     0.359523   
5093    AAIC         1.700000         6.690000                     0.254111   
5094    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  current_price  
0                     0.615231          60.23  
1                     0.363889          11.23  
2                     0.415323           3.32  
3                     0.477175          16.07  
4                     0.468899           1.40  
...                        ...            ...  
5090                  0.187663           3.72  
5091                  0.630388          17.90  
5092                  0.640477          18.01  
5093                  0.745889           3.69  
5094                  0.411684           1.57  

[5095 rows x 6 columns]

In [21]:
# dfcp1 = dfcp1.iloc[:, :-1]

In [22]:
# Calculation: We are trying to calculate how much the price has risen from the MINIMUM; the bottom of the crash in 27 March 2020
dfcp1['current_P_from_PostCrashInMarch2020_%'] = list(list(dfcp1['current_price'] - dfcp1['ticker_min_list'])/ dfcp1['ticker_min_list'])
dfcp1

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1       ZYXI         9.160000        14.400000                     0.636111   
2       ZYNE         2.900000         4.960000                     0.584677   
3       ZYME        24.280001        46.439999                     0.522825   
4        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
5090    AAME         2.120200         2.610000                     0.812337   
5091    AAMC        10.563500        28.580000                     0.369612   
5092     AAL        10.250000        28.510000                     0.359523   
5093    AAIC         1.700000         6.690000                     0.254111   
5094    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  current_price  \
0                     0.615231          60.23   
1                     0.363889          11.23   
2                     0.415323           3.32   
3                     0.477175          16.07   
4                     0.468899           1.40   
...                        ...            ...   
5090                  0.187663           3.72   
5091                  0.630388          17.90   
5092                  0.640477          18.01   
5093                  0.745889           3.69   
5094                  0.411684           1.57   

      current_P_from_PostCrashInMarch2020_%  
0                                  1.408237  
1                                  0.225983  
2                                  0.144828  
3                                 -0.338138  
4                                  0.261261  
...                                     ...  
5090                               0.754552  
5091                               0.694514  
5092                               0.757073  
5093                               1.170588  
5094                               1.156297  

[5095 rows x 7 columns]

In [23]:
# Calculation: We are trying to calculate how much the price has risen from the MAXIMUM; Pre-Crash of March 2020. 
dfcp1['current_P_from_PreCrashIn20Feb2020_%'] = list(list(dfcp1['current_price'] - dfcp1['ticker_max_list'])/ dfcp1['ticker_max_list'])
dfcp1 #I.E did it return to current levels?

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1       ZYXI         9.160000        14.400000                     0.636111   
2       ZYNE         2.900000         4.960000                     0.584677   
3       ZYME        24.280001        46.439999                     0.522825   
4        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
5090    AAME         2.120200         2.610000                     0.812337   
5091    AAMC        10.563500        28.580000                     0.369612   
5092     AAL        10.250000        28.510000                     0.359523   
5093    AAIC         1.700000         6.690000                     0.254111   
5094    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  current_price  \
0                     0.615231          60.23   
1                     0.363889          11.23   
2                     0.415323           3.32   
3                     0.477175          16.07   
4                     0.468899           1.40   
...                        ...            ...   
5090                  0.187663           3.72   
5091                  0.630388          17.90   
5092                  0.640477          18.01   
5093                  0.745889           3.69   
5094                  0.411684           1.57   

      current_P_from_PostCrashInMarch2020_%  \
0                                  1.408237   
1                                  0.225983   
2                                  0.144828   
3                                 -0.338138   
4                                  0.261261   
...                                     ...   
5090                               0.754552   
5091                               0.694514   
5092                               0.757073   
5093                               1.170588   
5094                               1.156297   

      current_P_from_PreCrashIn20Feb2020_%  
0                                -0.073385  
1                                -0.220139  
2                                -0.330645  
3                                -0.653962  
4                                -0.330144  
...                                    ...  
5090                              0.425287  
5091                             -0.373688  
5092                             -0.368292  
5093                             -0.448430  
5094                              0.268584  

[5095 rows x 8 columns]

- We can see in Z shot up by 140% recovering from 25.01 to 60.23 from the BOTTOM of the crash as shown in [current_P_from_PostCrashInMarch2020_]
- But it did not Recover to levels of pre-crash of price of 65.00. It missed by a shy of 7%

In [24]:
dfcp1.sort_values(by=['current_P_from_PostCrashInMarch2020_%'],ascending = False)

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
3897    CUBA         0.000000         6.524400                     0.000000   
4856    AMPG         0.030000         0.065500                     0.458015   
2284    MARA         0.396300         1.040000                     0.381058   
151      WLL         0.752300         2.850000                     0.263965   
3602     ELC         0.280000         0.400000                     0.700000   
...      ...              ...              ...                          ...   
2207    METX        10.000000        11.350000                     0.881057   
3637     EDU       103.849998       137.809998                     0.753574   
578     TLGT         2.250000         5.205000                     0.432277   
316      VER        19.900000        50.599998                     0.393281   
4612     BAK         3.860000        14.320000                     0.269553   

      value_lost_in_percentage  current_price  \
3897                  1.000000       5.410000   
4856                  0.541985       3.600000   
2284                  0.618942      40.439999   
151                   0.736035      68.300003   
3602                  0.300000      25.250000   
...                        ...            ...   
2207                  0.118943       0.295300   
3637                  0.246426       2.300000   
578                   0.567723       0.031300   
316                   0.606719            NaN   
4612                  0.730447            NaN   

      current_P_from_PostCrashInMarch2020_%  \
3897                                    inf   
4856                             119.000000   
2284                             101.043906   
151                               89.788251   
3602                              89.178571   
...                                     ...   
2207                              -0.970470   
3637                              -0.977853   
578                               -0.986089   
316                                     NaN   
4612                                    NaN   

      current_P_from_PreCrashIn20Feb2020_%  
3897                             -0.170805  
4856                             53.961832  
2284                             37.884615  
151                              22.964914  
3602                             62.124999  
...                                    ...  
2207                             -0.973982  
3637                             -0.983310  
578                              -0.993987  
316                                    NaN  
4612                                   NaN  

[5095 rows x 8 columns]

In [25]:
dfcp1.sort_values(by=['current_P_from_PreCrashIn20Feb2020_%'],ascending = False)

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
3602     ELC           0.2800         0.400000                     0.700000   
4856    AMPG           0.0300         0.065500                     0.458015   
2284    MARA           0.3963         1.040000                     0.381058   
3101     GME           3.3900         4.590000                     0.738562   
2744    IMCC           0.1000         0.100000                     1.000000   
...      ...              ...              ...                          ...   
2231    MDLX           2.8400        17.590000                     0.161455   
2232    MDLQ           2.7090        18.580000                     0.145802   
578     TLGT           2.2500         5.205000                     0.432277   
316      VER          19.9000        50.599998                     0.393281   
4612     BAK           3.8600        14.320000                     0.269553   

      value_lost_in_percentage  current_price  \
3602                  0.300000      25.250000   
4856                  0.541985       3.600000   
2284                  0.618942      40.439999   
3101                  0.261438     159.009995   
2744                  0.000000       3.440000   
...                        ...            ...   
2231                  0.838545       0.215000   
2232                  0.854198       0.215000   
578                   0.567723       0.031300   
316                   0.606719            NaN   
4612                  0.730447            NaN   

      current_P_from_PostCrashInMarch2020_%  \
3602                              89.178571   
4856                             119.000000   
2284                             101.043906   
3101                              45.905602   
2744                              33.400000   
...                                     ...   
2231                              -0.924296   
2232                              -0.920635   
578                               -0.986089   
316                                     NaN   
4612                                    NaN   

      current_P_from_PreCrashIn20Feb2020_%  
3602                             62.124999  
4856                             53.961832  
2284                             37.884615  
3101                             33.642699  
2744                             33.400000  
...                                    ...  
2231                             -0.987777  
2232                             -0.988428  
578                              -0.993987  
316                                    NaN  
4612                                   NaN  

[5095 rows x 8 columns]

In [26]:
snp500list = pd.read_csv('../dataset_orig/obtain_data_get_SNP500_List.csv',index_col = 0)
snp500list.head()
snp500list['snp500list'].to_list()[:5]

['MMM', 'ABT', 'ABBV', 'ABMD', 'ACN']

In [27]:
dfcp1

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1       ZYXI         9.160000        14.400000                     0.636111   
2       ZYNE         2.900000         4.960000                     0.584677   
3       ZYME        24.280001        46.439999                     0.522825   
4        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
5090    AAME         2.120200         2.610000                     0.812337   
5091    AAMC        10.563500        28.580000                     0.369612   
5092     AAL        10.250000        28.510000                     0.359523   
5093    AAIC         1.700000         6.690000                     0.254111   
5094    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  current_price  \
0                     0.615231          60.23   
1                     0.363889          11.23   
2                     0.415323           3.32   
3                     0.477175          16.07   
4                     0.468899           1.40   
...                        ...            ...   
5090                  0.187663           3.72   
5091                  0.630388          17.90   
5092                  0.640477          18.01   
5093                  0.745889           3.69   
5094                  0.411684           1.57   

      current_P_from_PostCrashInMarch2020_%  \
0                                  1.408237   
1                                  0.225983   
2                                  0.144828   
3                                 -0.338138   
4                                  0.261261   
...                                     ...   
5090                               0.754552   
5091                               0.694514   
5092                               0.757073   
5093                               1.170588   
5094                               1.156297   

      current_P_from_PreCrashIn20Feb2020_%  
0                                -0.073385  
1                                -0.220139  
2                                -0.330645  
3                                -0.653962  
4                                -0.330144  
...                                    ...  
5090                              0.425287  
5091                             -0.373688  
5092                             -0.368292  
5093                             -0.448430  
5094                              0.268584  

[5095 rows x 8 columns]

# SAVE CHECKPOINT 1

In [33]:
dfcp1.to_csv('Identify Largest Crash_1 (CheckPoint1).csv')

# Section 3: Track Stocks that are in S&P500
#### Rationale: 
Being large Cap Stocks-pro, in large volatile market, why subdue ourselves to so much stress with high volatility in small cap stocks and weird betas? So we just track the S&P500

# Checkpoint 2

In [28]:
df_snp500 = pd.DataFrame([ticker in snp500list['snp500list'].to_list() for ticker in dfcp1['colName'].to_list()],columns = ['S&P_500'])
dfcp2 = pd.concat([dfcp1, df_snp500],axis=1)
dfcp2

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
0          Z        25.010000        65.000000                     0.384769   
1       ZYXI         9.160000        14.400000                     0.636111   
2       ZYNE         2.900000         4.960000                     0.584677   
3       ZYME        24.280001        46.439999                     0.522825   
4        ZVO         1.110000         2.090000                     0.531101   
...      ...              ...              ...                          ...   
5090    AAME         2.120200         2.610000                     0.812337   
5091    AAMC        10.563500        28.580000                     0.369612   
5092     AAL        10.250000        28.510000                     0.359523   
5093    AAIC         1.700000         6.690000                     0.254111   
5094    AACG         0.728100         1.237600                     0.588316   

      value_lost_in_percentage  current_price  \
0                     0.615231          60.23   
1                     0.363889          11.23   
2                     0.415323           3.32   
3                     0.477175          16.07   
4                     0.468899           1.40   
...                        ...            ...   
5090                  0.187663           3.72   
5091                  0.630388          17.90   
5092                  0.640477          18.01   
5093                  0.745889           3.69   
5094                  0.411684           1.57   

      current_P_from_PostCrashInMarch2020_%  \
0                                  1.408237   
1                                  0.225983   
2                                  0.144828   
3                                 -0.338138   
4                                  0.261261   
...                                     ...   
5090                               0.754552   
5091                               0.694514   
5092                               0.757073   
5093                               1.170588   
5094                               1.156297   

      current_P_from_PreCrashIn20Feb2020_%  S&P_500  
0                                -0.073385    False  
1                                -0.220139    False  
2                                -0.330645    False  
3                                -0.653962    False  
4                                -0.330144    False  
...                                    ...      ...  
5090                              0.425287    False  
5091                             -0.373688    False  
5092                             -0.368292     True  
5093                             -0.448430    False  
5094                              0.268584    False  

[5095 rows x 9 columns]

In [29]:
dfcp2_snp500 = dfcp2[dfcp2['S&P_500'] == True]
dfcp2_snp500

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
7        ZTS        92.660004       143.619995                     0.645175   
21      ZION        24.080000        46.250000                     0.520649   
32      ZBRA       161.509995       242.720001                     0.665417   
33       ZBH        80.629997       158.389999                     0.509060   
37       YUM        56.520000       103.870003                     0.544142   
...      ...              ...              ...                          ...   
5077     ABC        73.930000        97.000000                     0.762165   
5081    ABBV        64.500000        94.959999                     0.679233   
5086     AAP        75.029999       142.479996                     0.526600   
5087    AAPL        56.092499        80.074997                     0.700500   
5092     AAL        10.250000        28.510000                     0.359523   

      value_lost_in_percentage  current_price  \
7                     0.354825     231.449997   
21                    0.479351      63.209999   
32                    0.334583     614.549988   
33                    0.490940     124.489998   
37                    0.455858     133.869995   
...                        ...            ...   
5077                  0.237835     123.129997   
5081                  0.320767     125.470001   
5086                  0.473400     240.039993   
5087                  0.299500     179.449997   
5092                  0.640477      18.010000   

      current_P_from_PostCrashInMarch2020_%  \
7                                  1.497841   
21                                 1.625000   
32                                 2.805028   
33                                 0.543966   
37                                 1.368542   
...                                     ...   
5077                               0.665494   
5081                               0.945271   
5086                               2.199254   
5087                               2.199180   
5092                               0.757073   

      current_P_from_PreCrashIn20Feb2020_%  S&P_500  
7                                 0.611544     True  
21                                0.366703     True  
32                                1.531930     True  
33                               -0.214029     True  
37                                0.288822     True  
...                                    ...      ...  
5077                              0.269381     True  
5081                              0.321293     True  
5086                              0.684728     True  
5087                              1.241024     True  
5092                             -0.368292     True  

[492 rows x 9 columns]

Need to check why only 492 rows, prob wikipedia not updated but thats fine, taking a quick look

In [30]:
dfcp2_snp500.sort_values(by='current_P_from_PostCrashInMarch2020_%',ascending=False)

colName  ticker_min_list  ticker_max_list  Low_from_High_in_percentage  \
2075    MRNA        18.230000        31.580000                     0.577264   
496     TSLA        72.244003       180.199997                     0.400910   
3836     CZR         7.100000        69.470001                     0.102202   
1520    PENN         4.520000        38.169998                     0.118418   
3565    ENPH        23.990000        59.090000                     0.405991   
...      ...              ...              ...                          ...   
4496    BIIB       268.880005       336.709991                     0.798551   
444        T        26.770000        38.610001                     0.693344   
4832    ANET       157.039993       228.279999                     0.687927   
3898    CTXS       102.970001       137.809998                     0.747188   
3925     CSX        47.660000        80.449997                     0.592418   

      value_lost_in_percentage  current_price  \
2075                  0.422736     257.059998   
496                   0.599090    1017.030029   
3836                  0.897798      88.980003   
1520                  0.881582      48.959999   
3565                  0.594009     215.020004   
...                        ...            ...   
4496                  0.201449     232.619995   
444                   0.306656      22.840000   
4832                  0.312073     130.690002   
3898                  0.252812      81.510002   
3925                  0.407582      36.740002   

      current_P_from_PostCrashInMarch2020_%  \
2075                              13.100933   
496                               13.077709   
3836                              11.532395   
1520                               9.831858   
3565                               7.962901   
...                                     ...   
4496                              -0.134856   
444                               -0.146806   
4832                              -0.167792   
3898                              -0.208410   
3925                              -0.229123   

      current_P_from_PreCrashIn20Feb2020_%  S&P_500  
2075                              7.139962     True  
496                               4.643896     True  
3836                              0.280841     True  
1520                              0.282683     True  
3565                              2.638856     True  
...                                    ...      ...  
4496                             -0.309138     True  
444                              -0.408443     True  
4832                             -0.427501     True  
3898                             -0.408533     True  
3925                             -0.543319     True  

[492 rows x 9 columns]

In [31]:
from pandas import set_option
pd.set_option('max_rows',500)

In [32]:
dfcp2_snp500.sort_values(by='current_P_from_PostCrashInMarch2020_%',ascending=False).reset_index()

index colName  ticker_min_list  ticker_max_list  \
0     2075    MRNA        18.230000        31.580000   
1      496    TSLA        72.244003       180.199997   
2     3836     CZR         7.100000        69.470001   
3     1520    PENN         4.520000        38.169998   
4     3565    ENPH        23.990000        59.090000   
5     3687     DVN         5.410000        22.150000   
6     3391     FCX         5.310000        12.080000   
7     3423    FANG        15.560000        78.480003   
8     3485    ETSY        31.690001        62.459999   
9     4813     APA         4.310000        28.580000   
10    2189     MGM         7.140000        32.340000   
11     939    SIVB       130.979996       270.790009   
12    3210       F         4.010000         8.030000   
13    2100     MOS         7.010000        19.410000   
14    2997     HAL         4.610000        21.950001   
15    4915     ALB        50.900002        94.309998   
16    4792    APTV        33.549999        91.089996   
17     378     URI        69.589996       155.720001   
18    4908    ALGN       137.720001       272.489990   
19    2073     MRO         3.350000        10.250000   
20    1316     PWR        23.799999        40.240002   
21    3791     DFS        25.250000        75.790001   
22    3722     DRI        34.160000       121.099998   
23    3090    GNRC        79.010002       118.080002   
24    3234    FTNT        75.699997       118.000000   
25    2379     LKQ        13.900000        34.020000   
26     543     TPR        10.580000        28.809999   
27     528    TRMB        21.320000        46.130001   
28    1691     OKE        15.370000        77.519997   
29    4877    AMAT        37.990002        66.400002   
30    2344     LOW        65.019997       126.570000   
31    1791     NUE        28.420000        48.570000   
32    2416     LEN        29.350000        71.029999   
33    2363     LNC        17.059999        61.029999   
34    2637      IT        83.239998       155.460007   
35    2093     MPC        16.620001        60.259998   
36    3780     DHI        28.780001        62.049999   
37      32    ZBRA       161.509995       242.720001   
38    2296     LYV        29.500000        75.660004   
39    4663    AVGO       167.869995       309.239990   
40    3098      GM        16.799999        35.290001   
41    2331    LRCX       187.889999       328.489990   
42    3341    FITB        11.670000        29.830000   
43     701     SYF        12.800000        33.639999   
44    2090    MPWR       135.220001       184.059998   
45    1009     SEE        17.549999        34.970001   
46    4082     CMG       465.209991       927.590027   
47    3441    EXPE        45.650002       121.660004   
48    3971     CRL        98.900002       175.429993   
49     625     TER        44.290001        67.889999   
50    2518    KLAC       114.019997       167.300003   
51    4853     AMP        82.879997       179.720001   
52     170     WHR        64.220001       149.080002   
53    4870     AMD        38.709999        57.270000   
54    2978     HCA        68.129997       148.110001   
55    2052      MS        27.809999        53.750000   
56    4033     COF        42.869999       103.540001   
57    1749    NXPI        64.559998       134.800003   
58    1320     PVH        29.049999        86.879997   
59    2699    INTU       194.720001       301.239990   
60     452     TXT        21.660000        47.630001   
61     501    TSCO        67.360001        99.779999   
62    4275    CBRE        29.830000        63.700001   
63     840     SPG        44.919998       142.250000   
64    4237    CDNS        54.320000        76.970001   
65    1245     RCL        22.330000       109.800003   
66    2505     KMX        44.270000       101.900002   
67    2617     JCI        23.500000        42.450001   
68    2831     HWM         9.190345        25.831078   
69    1400    POOL       167.779999       232.369995   
70     117     WST       128.

# To Do: Market Cap

# To Do: PE Ratio

# How to Analyze